examples/speed_test_async_multi.cpp

```diff
@@ -91,30 +103,32 @@ int main(int argc, char **argv) {
     // wrap it into bredis connection
     r::Connection<next_layer_t> c(std::move(socket));
 
-    Buffer tx_buff, rx_buff;
-    std::promise<std::string> completion_promise;
-    std::future<std::string> completion_future =
-        completion_promise.get_future();
+    std::string tx_backend, rx_backend;
+    //tx_backend.reserve(cmds_count * 4);
+    //rx_backend.reserve(cmds_count * 4);
+    Buffer tx_buff(tx_backend), rx_buff(rx_backend);
+    //Buffer tx_buff, rx_buff;
+    std::promise<void> completion_promise;
+    auto completion_future = completion_promise.get_future();
 
     c.async_read(
         rx_buff,
-        [&](const boost::system::error_code &ec, auto &&r) {
+        [&](const sys::error_code &ec, result_t &&r) {
             assert(!ec);
-            auto &replies =
-                boost::get<r::markers::array_holder_t<Iterator>>(r.result);
-            auto &last_reply = replies.elements.at(replies.elements.size() - 1);
-            auto &str_reply =
-                boost::get<r::markers::string_t<Iterator>>(last_reply);
-            std::string value{str_reply.from, str_reply.to};
+            (void)ec;
```

Comment:
Why "ec" should be converted to void? and also there are better ways for casting to void in c++